### PR TITLE
Add Remote/face to face advice method logic

### PR DESCRIPTION
**Pull request by neoeno**
This will make more sense when I do the PR on RAD. [EDIT: https://github.com/moneyadviceservice/rad/pull/275]

Also sorry for the rename commit — need to work on my TTD (time-travel driven development)

### CODE REVIEW

File: app/models/firm.rb [hunk 1]
```diff
@@ -151,6 +165,11 @@ def advice_types
     ADVICE_TYPES_ATTRIBUTES.map { |a| [a, self[a]] }.to_h
   end
 
+  def primary_advice_method
```
Comment (tomas-stefano) [inline app/models/firm.rb]: Where this method is called? Is only on the validation?

Comment (tomas-stefano) [inline app/models/firm.rb]: If it is only on the validation, can we create a custom validator for the case?

Comment (neoeno) [inline app/models/firm.rb]: It's used by RAD too — see here for instance: https://github.com/moneyadviceservice/rad/pull/275/files#diff-6004437aeea98d71ab51826159b8414fR8

Nice idea though — not used custom validators before! Will be sure to in future as big messy validations are one of my least favourite things.

**Comment by neoeno** [comment]
@rnalexander Thanks! I fixed that in a later commit but as you rightly pointed out I can squash it from the history entirely, so I've done that.

@JiggyPete Good idea on the `if:` front. The syntax ended up being slightly different but I've implemented the spirit of it.

**Comment by JiggyPete** [comment]
:+1:

File: app/models/firm.rb (hunk 1 above)
Comment (tomas-stefano) [inline app/models/firm.rb]: Ok. Then consider next time in creating custom validators. Totally worth it. :)

**Comment by tomas-stefano** [comment]
:+1:

**Comment by rnalexander** [comment]
:+1: :shipit:

**Pull request merged**